### PR TITLE
Add  Independent and Instance Buffer mesh playground

### PR DESCRIPTION
**Pull request by GuoLei1990**
Add Independent and Instance Buffer mesh playground

### CODE REVIEW

File: playground/buffer-mesh-independent.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  const posNorBuffer = new Buffer(engine, BufferBindFlag.VertexBuffer, positionNormals, BufferUsage.Static);
+  const independentColorBuffer = new Buffer(engine, BufferBindFlag.VertexBuffer, colorData, BufferUsage.Dynamic);
+  const indexBuffer = new Buffer(engine, BufferBindFlag.IndexBuffer, indices, BufferUsage.Static);
+
+  // Bind buffer
+  cubeMesh.setVertexBufferBinding(posNorBuffer, 24, 0);
+  cubeMesh.setVertexBufferBinding(independentColorBuffer, 12, 1);
+  cubeMesh.setIndexBufferBinding(indexBuffer, IndexFormat.UInt16);
+
+  // Add vertexElement
```
Comment (cptbtptpbcptdtptp) [inline playground/buffer-mesh-independent.ts:152]: Add vertexElement.

File: playground/buffer-mesh-independent.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+          16, 18, 17, 18, 16, 19,
+          // Back
+          20, 22, 23, 22, 20, 21]);
+
+  // Create gpu vertex buffer and index buffer.
+  const posNorBuffer = new Buffer(engine, BufferBindFlag.VertexBuffer, positionNormals, BufferUsage.Static);
+  const independentColorBuffer = new Buffer(engine, BufferBindFlag.VertexBuffer, colorData, BufferUsage.Dynamic);
+  const indexBuffer = new Buffer(engine, BufferBindFlag.IndexBuffer, indices, BufferUsage.Static);
+
+  // Bind buffer
```
Comment (cptbtptpbcptdtptp) [inline playground/buffer-mesh-independent.ts:147]: Bind buffer.

File: playground/buffer-mesh-independent.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  PointLight,
+  Vector3,
+  VertexElement,
+  VertexElementFormat,
+  WebGLEngine,
+  Script
+} from "oasis-engine";
+
+/**
+ * Script for updating ducks position, rotation, and scale.
```
Comment (cptbtptpbcptdtptp) [inline playground/buffer-mesh-independent.ts:25]: Incorrect comment.

File: playground/buffer-mesh-independent.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+ */
+class RandomColorScript extends Script {
+  /** Color data. */
+  colorData: Float32Array;
+  /** Color buffer. */
+  colorBuffer: Buffer;
+
+  private _loopCount = 0;
+
```
Comment (cptbtptpbcptdtptp) [inline playground/buffer-mesh-independent.ts:34]: Does not show the effect of color change.

**Review by cptbtptpbcptdtptp — CHANGES_REQUESTED** [review]

File: playground/buffer-mesh-independent.ts (hunk 1 above)
Comment (GuoLei1990) [reply playground/buffer-mesh-independent.ts:152]: done

File: playground/buffer-mesh-independent.ts (hunk 2 above)
Comment (GuoLei1990) [reply playground/buffer-mesh-independent.ts:147]: done

File: playground/buffer-mesh-independent.ts (hunk 3 above)
Comment (GuoLei1990) [reply playground/buffer-mesh-independent.ts:25]: done

File: playground/buffer-mesh-independent.ts (hunk 4 above)
Comment (GuoLei1990) [reply playground/buffer-mesh-independent.ts:34]: Need merge this https://github.com/oasis-engine/engine/pull/450 for engine.

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Pull request merged**